ignition-server/controllers/tokensecret_controller.go

```diff
@@ -274,12 +281,24 @@ func (r *TokenSecretReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return payload, err
 	}()
 	if err != nil {
-		patch := tokenSecret.DeepCopy()
-		patch.Data[TokenSecretReasonKey] = []byte(InvalidConfigReason)
-		patch.Data[TokenSecretMessageKey] = []byte(fmt.Sprintf("Failed to generate payload: %s", err))
-		if err := r.Client.Patch(ctx, patch, client.MergeFrom(tokenSecret)); err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to patch tokenSecret with payload content: %w", err)
+		// This patch could flood the API server, so we should only do it when the reason or message is different from the current one.
+		// More info here: https://issues.redhat.com/browse/OCPBUGS-42320.
+		if !hasSameReasonAndMessage(tokenSecret, InvalidConfigReason, fmt.Sprintf("Failed to generate payload: %s", err)) {
```

Comment:
nit: can we instead return early if hasSameReasonAndMessage, and otherwise run the bulk logic